From 98419a4133f6358ec2e8ad56c62aacf3e3754c3e Mon Sep 17 00:00:00 2001
From: Peng Huang <shawn.p.huang@gmail.com>
Date: Wed, 8 Dec 2010 16:37:42 +0800
Subject: [PATCH] Fix GI transfer mode annotation in ibus_bus_list_*engines() comment.

See https://bugzilla.gnome.org/show_bug.cgi?id=635248.  Also, do not mark those
functions as "not implemented", since they are apparently implemented.

BUG=none
TEST=manual

Review URL: http://codereview.appspot.com/3274044
---
 src/ibusbus.h |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/ibusbus.h b/src/ibusbus.h
index 2e288f5..fb56b76 100644
--- a/src/ibusbus.h
+++ b/src/ibusbus.h
@@ -246,21 +246,18 @@ gboolean     ibus_bus_register_component(IBusBus        *bus,
 /**
  * ibus_bus_list_engines:
  * @bus: An IBusBus.
- * @returns: (transfer full) (element-type IBusEngineDesc): A List of engines.
+ * @returns: (transfer container) (element-type IBusEngineDesc): A List of engines.
  *
  * List engines.
- * Note that this function is not yet implemented.
  */
 GList       *ibus_bus_list_engines      (IBusBus        *bus);
 
 /**
  * ibus_bus_list_active_engines:
  * @bus: An IBusBus.
- * @returns: (transfer full) (element-type IBusEngineDesc): A List of active engines.
+ * @returns: (transfer container) (element-type IBusEngineDesc): A List of active engines.
  *
  * List active engines.
- * Note that this function is not yet implemented.
- * <note><para>Not yet implemented.</para></note>
  */
 GList       *ibus_bus_list_active_engines
                                         (IBusBus        *bus);
-- 
1.7.3.2

--- a/src/ibusinputcontext.c
+++ b/src/ibusinputcontext.c
@@ -732,7 +732,7 @@ ibus_input_context_get_input_context (co
                                       GDBusConnection     *connection)
 {
     IBusInputContext *context;
-    GError *error;
+    GError *error = NULL;
 
     context = ibus_input_context_new (path, connection, NULL, &error);
     if (!context) {
@@ -904,7 +904,7 @@ ibus_input_context_is_enabled (IBusInput
 {
     g_assert (IBUS_IS_INPUT_CONTEXT (context));
     GVariant *result;
-    GError *error;
+    GError *error = NULL;
     result = g_dbus_proxy_call_sync ((GDBusProxy *) context,
                                      "IsEnabled",               /* method_name */
                                      NULL,                      /* parameters */
@@ -932,7 +932,7 @@ ibus_input_context_get_engine (IBusInput
 {
     g_assert (IBUS_IS_INPUT_CONTEXT (context));
     GVariant *result;
-    GError *error;
+    GError *error = NULL;
     result = g_dbus_proxy_call_sync ((GDBusProxy *) context,
                                      "GetEngine",               /* method_name */
                                      NULL,                      /* parameters */
From 017077ceb9ec2f26a8c524f3794a832164f21768 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@unixuser.org>
Date: Tue, 28 Dec 2010 12:46:25 +0900
Subject: [PATCH] Fix g_variant_get() call in DeleteSurroundingText signal handler.

BUG=none
TEST=manual

Review URL: http://codereview.appspot.com/3820042
---
 src/ibusinputcontext.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ibusinputcontext.c b/src/ibusinputcontext.c
index 88afc22..7bbf8a2 100644
--- a/src/ibusinputcontext.c
+++ b/src/ibusinputcontext.c
@@ -578,7 +578,7 @@ ibus_input_context_g_signal (GDBusProxy  *proxy,
         guint32 keycode;
         guint32 state;
 
-        g_variant_get (parameters, 0, "(uuu)", &keyval, &keycode, &state);
+        g_variant_get (parameters, "(uuu)", &keyval, &keycode, &state);
 
         /* Forward key event back with IBUS_FORWARD_MASK. And process_key_event will
          * not process key event with IBUS_FORWARD_MASK again. */
@@ -595,7 +595,7 @@ ibus_input_context_g_signal (GDBusProxy  *proxy,
         gint offset_from_cursor;
         guint nchars;
 
-        g_variant_get (parameters, 0, "(iu)", &offset_from_cursor, &nchars);
+        g_variant_get (parameters, "(iu)", &offset_from_cursor, &nchars);
 
         g_signal_emit (context,
                        context_signals[DELETE_SURROUNDING_TEXT],
-- 
1.7.3.2

